zebra/zebra_vty.c

```diff
@@ -151,27 +340,31 @@ static int zebra_static_route_leak(struct vty *vty,
 	memset(&snh_label, 0, sizeof(struct static_nh_label));
 	if (label_str) {
 		if (!mpls_enabled) {
-			vty_out(vty,
-				"%% MPLS not turned on in kernel, ignoring command\n");
+			if (vty)
+				vty_out(vty,
+					"%% MPLS not turned on in kernel, ignoring command\n");
 			return CMD_WARNING_CONFIG_FAILED;
 		}
 		int rc = mpls_str2label(label_str, &snh_label.num_labels,
 					snh_label.label);
 		if (rc < 0) {
 			switch (rc) {
 			case -1:
-				vty_out(vty, "%% Malformed label(s)\n");
+				if (vty)
+					vty_out(vty, "%% Malformed label(s)\n");
 				break;
 			case -2:
-				vty_out(vty,
-					"%% Cannot use reserved label(s) (%d-%d)\n",
-					MPLS_LABEL_RESERVED_MIN,
-					MPLS_LABEL_RESERVED_MAX);
+				if (vty)
+					vty_out(vty,
+						"%% Cannot use reserved label(s) (%d-%d)\n",
+						MPLS_LABEL_RESERVED_MIN,
+						MPLS_LABEL_RESERVED_MAX);
 				break;
```

Comment:
that event should be logged into a zlog_xxx

Reply:
dibe